File: python/stdlib/default/splitflow5.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("Splitflow 5", "default.splitflow5", "", {"val": "Object"},
                                   {"1": "Object", "2": "Object", "3": "Object", "4": "Object", "5": "Object"},
                                   "Splits the flow.", verbose)

    def tick(self, value):
        tags = {}
        if "tags" in value and "val" in value["tags"]:
            tag = value["tags"]["val"]
            tags = {"1": tag, "2": tag, "3": tag, "4": tag, "5": tag}
        return {"tags": tags, "1": value["val"], "2": value["val"], "3": value["val"], "4": value["val"], "5": value["val"]}

File: lua/stdlib/default/splitflow5.lua
local myNode = {}
function myNode.init(node)
  node.tick = function(value)
    result = {}
    result["1"] = value.val
    result["2"] = value.val
    result["3"] = value.val
    result["4"] = value.val
    result["5"] = value.val
    return {}
  end
end

function myNode.spec(node)
  node.name = "Splitflow 5"
  node.inputs.val = "Object"
  node.outputs["1"] = "Object"
  node.outputs["2"] = "Object"
  node.outputs["3"] = "Object"
  node.outputs["4"] = "Object"
  node.outputs["5"] = "Object"
  node.desc = "Splits the flow into 5."
end

return myNode
